### DISCUSSION TITLE
Gitpod flashing process understanding

### DISCUSSION THREAD

**Discussion by jeffbrown57** [arendst/Tasmota#21404] [Q&A]
sorry if this is answered elsewhere , i certainly didn;t find it. So i have flashed an esp32 for first use with gitpod , selecting tasmota32 , and all is well. now i want to add an AHT2x temp/hum sensor to it. , i assume i delete .cache, .pio   recompile it
then use the tasmota.bin ( not tasmota.factory.bin) and then flash it with "firmware upgrade" from the Console. Is that the correct
way ?

**Comment by barbudor** [comment]
Not from the console
You first need to download to your computer the 'tasmota.bin' file from gitpod
Then from the menu 'upgrade firmware', select 'upgrade from file' (bottom part) using the tasmota.bin file

**Reply by jeffbrown57** [reply]
thanks for responding , for clarification I have previously flashed with tasmota.factory.bin , then subsequently i add a sensor
by editing the override.h file adding '#define USE_SENSOR'  after compiling is complete i ignore the newer tasmota.factory.bin just built , download the newly compiled tasmota.bin and then use that file selecting 'upgrade firmware' from that ,

is that about right ?

**Reply by Jason2866** [reply]
After initial flash with the factory firmware every update is done with the plain firmware

**Comment by jeffbrown57** [comment]
Thanks for the update !!
